src/jit/gentree.cpp

```diff
@@ -6788,6 +6792,27 @@ GenTreeArgList* Compiler::gtNewArgList(GenTreePtr arg1, GenTreePtr arg2, GenTree
     return new (this, GT_LIST) GenTreeArgList(arg1, gtNewArgList(arg2, arg3));
 }
 
+/*****************************************************************************
+*
+*  Create a list out of the four values.
+*/
+
+GenTreeArgList* Compiler::gtNewArgList(GenTreePtr arg1, GenTreePtr arg2, GenTreePtr arg3, GenTreePtr arg4)
+{
+    return new (this, GT_LIST) GenTreeArgList(arg1, gtNewArgList(arg2, arg3, arg4));
+}
+
+/*****************************************************************************
+*
+*  Create a list out of the five values.
+*/
+
+GenTreeArgList* Compiler::gtNewArgList(
```

Comment:
Do we need this? Parameter counts above 4 looked to be very uncommon, so a helper function that calls the smaller ones and works with arbitrary sizes might be more useful

Reply:
One of my experiments to be removed - when you run a loop on 8 or 16 arguments and use in the first step 4 args function in every next step you need 5 args function to pass 4 args and append them to 1 list - but actually function quite different from this one.